### ISSUE TITLE
Error messages are not descriptive

### ISSUE THREAD

**Issue by jose-cds** [cds-snc/platform-forms-client#3085]
**Health Canada Label: Serious**

Pages:  Advancing Accessibility Standards Research Program Expression of Interest
URL: https://forms-staging.cdssandbox.xyz/en/id/134
User agent:  Chrome (v119.0.6045.200), MS Edge (120.0.2210.77),  JAWS2020, NVDA 2023.1
Language: English and French

Issue Description: 
In the error summary provided above the form, the error messages for leaving the required form fields empty are the same for all the fields. The label text associated with the field in error is not identified in the corresponding error message in the summary. In this case, the error message doesn’t provide sufficient context for users to understand what field is in error.
In addition, when the form is submitted leaving the required fields empty, all the error messages are announced by screen readers at once as an alert. It can be confusing for screen reader users.

<div class="gc-alert gc-alert--error gc-alert--validation" data-testid="alert" id="gc-form-errors" tabindex="0" role="alert"><div class="gc-alert__body"><h2>Please correct the errors on the page.</h2> 

Steps to reproduce:

1. Run the NVDA screen reader
2. Open the above URL in the Chrome browser
3. Navigate using the "Tab" key
4. Leave some required fields empty
5. Activate the the "Sumbit" button
6. Listen to the screen reader 

Expected result: The error message should be descriptive and concise to provide enough context to the users to clearly understand it.
Please refer to the WET Working example: Form validation as a reference: https://wet-boew.github.io/v4.0-ci/demos/formvalid/formvalid-en.html

**Comment by srtalbot** [comment]
Duplicate card, please see: https://app.zenhub.com/workspaces/gcforms-60cb8929764d71000e481cab/issues/gh/cds-snc/platform-forms-client/3950
